packages/neon-core/src/tx/components/Cosigner.ts

```diff
@@ -0,0 +1,90 @@
+import { WitnessScope, extractScopes } from "./WitnessScope";
+import { StringStream, num2hexstring } from "../../u";
+import { deserializeArrayOf, serializeArrayOf } from "../main";
+
+export interface CosignerLike {
+  account: string;
+  scopes: WitnessScope[];
+  allowedContracts: string[];
+  allowedGroups: string[];
+}
+
+function isTargetContainedInScopes(
+  scopes: WitnessScope[],
+  targetScope: WitnessScope
+): boolean {
+  if (targetScope === WitnessScope.Global) {
+    return scopes.every(scope => scope === WitnessScope.Global);
+  }
+  return scopes.includes(targetScope);
+}
+
+export class Cosigner {
+  /**
+   * script hash of cosigner
+   */
+  public account: string;
+  public scopes: WitnessScope[];
+  public allowedContracts: string[];
+  public allowedGroups: string[];
```

Comment:
We can start thinking about typing these fields more strictly in the future.

Reply:
I had this thought, too.